src/utils.ts

```diff
@@ -12,38 +12,256 @@
+import {
+  DataTypeMap,
+  logicalNot,
+  tensor,
+  Tensor,
+  Tensor1D,
+  tensor1d,
+  Tensor2D,
+  tensor2d,
+  TensorLike,
+  tidy,
+  where,
+  zerosLike,
+} from '@tensorflow/tfjs-node'
+import { DataFrame, Series } from 'danfojs-node'
+import { ArrayType1D, ArrayType2D } from 'types'
+
+export type Scikit1D = TensorLike1D | Tensor1D | Series
+export type Scikit2D = TensorLike2D | Tensor2D | DataFrame
+export type ScikitVecOrMatrix = Scikit1D | Scikit2D
 
 /**
-* Generates an array of dim (row x column) with inner values set to zero
-* @param row 
-* @param column 
-*/
-export const zeros = (row: number, column: number): ArrayType1D | ArrayType2D => {
-    const zeroData = [];
-    for (let i = 0; i < row; i++) {
-        const colData = Array(column);
-        for (let j = 0; j < column; j++) {
-            colData[j] = 0;
-        }
-        zeroData.push(colData);
+ * Generates an array of dim (row x column) with inner values set to zero
+ * @param row
+ * @param column
+ */
+export const zeros = (
+  row: number,
+  column: number
+): ArrayType1D | ArrayType2D => {
+  const zeroData = []
+  for (let i = 0; i < row; i++) {
+    const colData = Array(column)
+    for (let j = 0; j < column; j++) {
+      colData[j] = 0
     }
-    return zeroData;
+    zeroData.push(colData)
+  }
+  return zeroData
 }
 
-
 /**
  * Checks if array is 1D
- * @param arr The array 
-*/
+ * @param arr The array
+ */
 export const is1DArray = (arr: ArrayType1D | ArrayType2D): boolean => {
-    if (
-        typeof arr[0] == "number" ||
-        typeof arr[0] == "string" ||
-        typeof arr[0] == "boolean"
-    ) {
-        return true;
+  if (
+    typeof arr[0] == 'number' ||
+    typeof arr[0] == 'string' ||
+    typeof arr[0] == 'boolean'
+  ) {
+    return true
+  } else {
+    return false
+  }
+}
+
+/**
+ *
+ * @param data Scikit1D One dimensional array of data
+ * @returns Tensor1D. If you pass in something that isn't 1D, then it will throw an error.
+ * This is the case with 2D Tensors as well. If you really want to reshape them then use tf.reshape
+ */
+export function convertToTensor1D(data: Scikit1D, dtype?: DataType): Tensor1D {
+  if (data instanceof Series) {
+    // Do type inference if no dtype is passed, otherwise try to parse as that dtype
+    return dtype ? tensor1d(data.values, dtype) : tensor1d(data.values)
+  }
+  if (data instanceof Tensor) {
+    if (data.shape.length === 1) {
+      if (!dtype || data.dtype == dtype) {
+        return data
+      }
+      return data.asType(dtype)
+    } else {
+      throw new Error(
+        'ParamError: if data is a Tensor it must be a Tensor1D. If you really meant to reshape this tensor than use tf.reshape'
+      )
+    }
+  }
+  return dtype ? tensor1d(data, dtype) : tensor1d(data)
+}
+
+export function convertToTensor2D(data: Scikit2D, dtype?: DataType): Tensor2D {
+  if (data instanceof DataFrame) {
+    return dtype
+      ? tensor2d(data.values, undefined, dtype)
```

Comment:
Just a tip, you can get the 2D tensor from a DataFrame directly. E.g:

`DataFrame.tensor`